src/components/message/message-card/MessageCardBase.tsx

```diff
@@ -17,106 +17,142 @@
+		isExported,
+		isExport,
+		isExpanded,
 		isAttached,
 		isBookmarked,
 		toogleMessagePin,
 		isEmbedded,
 		sortOrderItems,
 	}: MessageCardBaseProps) => {
-		const { id, rawMessageBase64, parsedMessage } = message;
-
-		const bookmarkIconClass = createBemBlock('bookmark-button', isBookmarked ? 'pinned' : 'hidden');
-
-		const toggleViewType = (v: MessageViewType) => {
-			setViewType(v);
-		};
+		const { id, rawMessageBase64 } = message;
 
-		const isScreenshotMsg = isScreenshotMessage(message);
+		const rootClass = createBemBlock(
+			'message-card-wrapper',
+			isAttached ? 'attached' : null,
+			isBookmarked ? 'pinned' : null,
+			isHighlighted ? 'highlighted' : null,
+			isSoftFiltered ? 'soft-filtered' : null,
+			isExport ? 'export-mode' : null,
+			isExported ? 'exported' : null,
+		);
 
 		const messageViewTypeRendererProps: MessageCardViewTypeRendererProps = {
-			viewType,
 			messageId: id,
-			messageBody: parsedMessage ? parsedMessage.message : null,
-			isBeautified: viewType === MessageViewType.FORMATTED,
 			rawContent: rawMessageBase64,
 			isSelected: isAttached || false,
 			sortOrderItems: sortOrderItems || [],
 		};
 
-		const messageCardToolsConfig: MessageCardToolsConfig = {
+		const messageCardToolsConfig: MessageCardToolsProps = {
 			message,
-			parsedMessage,
-			messageViewType: viewType,
-			toggleViewType,
 			isBookmarked: isBookmarked || false,
 			toggleMessagePin: toogleMessagePin || (() => null),
-			isScreenshotMsg,
 			isEmbedded,
 		};
 
+		const messageInfoProps: MessageInfoProps = {
+			message,
+			viewType: viewTypes ? viewTypes[0] : viewType,
+			setViewType,
+			getSavedViewType,
+			onTimestampMouseEnter: hoverMessage,
+			onTimestampMouseLeave: unhoverMessage,
+			isBookmarked: isBookmarked || false,
+			isEmbedded,
+			isScreenshotMsg: isScreenshotMessage(message.parsedMessages?.[0]),
+		};
+
 		return (
-			<div>
-				{!isEmbedded && isBookmarked && <div className={bookmarkIconClass} />}
+			<div className={rootClass} onClick={addMessagesToExport}>
 				<div className='message-card'>
-					<div className='mc__mc-body mc-body'>
-						<MessageHeader
-							message={message}
-							parsedMessage={parsedMessage}
-							onTimestampMouseEnter={hoverMessage}
-							onTimestampMouseLeave={unhoverMessage}
-						/>
-						{isScreenshotMsg ? (
-							<div className='mc-body__screenshot'>
-								<MessageScreenshotZoom
-									src={
-										typeof rawMessageBase64 === 'string'
-											? `data:${parsedMessage?.message.metadata.messageType};base64,` +
-											  `${message.rawMessageBase64}`
-											: ''
-									}
-									alt={message.id}
-								/>
-							</div>
-						) : (
-							<div className='mc-body__human'>
-								<MessageCardViewTypeRenderer {...messageViewTypeRendererProps} />
+					<MessageHeader {...messageCardToolsConfig} {...messageInfoProps} />
+					{message.parsedMessages
+						?.filter((parsedMessage, index) => (isExpanded ? parsedMessage : index < 1))
```

Comment:
what does ' index < 1' do ? why dont we just return nothing if it's not expanded like so
`{isExpanded && parsedMessages.map()}`

Reply:
if its not expanded, we still need to return first element, like on pic, or is it old version?
![image](https://user-images.githubusercontent.com/52595110/177712267-131aeca9-9e67-4b24-88be-1978f921a1f2.png)